main/src/cgeo/geocaching/calculator/CoordinatesCalculateUtils.java

```diff
@@ -0,0 +1,163 @@
+package cgeo.geocaching.calculator;
+
+import cgeo.geocaching.models.CalcState;
+import cgeo.geocaching.settings.Settings;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * Through out this implementation:
+ *
+ * 'Equations' are used to represent 'Variables' that appear in the description of the cache coordinated themselves.
+ *             As in "N 42° 127.ABC".  In this example 'A', 'B' and 'C' are all 'equations'.
+ *             All 'equations' must have a CAPITAL-LETTER name.
+ *
+ * 'FreeVariables' are used to represent 'Variables' that appear in the 'expression' of an equation
+ *                 As in "X = a^2 + b^2".  In this example 'a' and 'b' are both 'freeVariables'.
+ *                 All 'freeVariables' must have a lower-case name.
+ */
+public final class CoordinatesCalculateUtils {
+
+    /** Flag values used to designate that no AutoChar has been set */
+    public static final char EMPTY_CHAR = '-';
+
+    private CoordinatesCalculateUtils() {
+        // Do not instantiate
+    }
+
+    public static List<VariableData> sortVariables(final List<VariableData> variables,
+                                                   final List<VariableData> varBank,
+                                                   final String variableNames,
+                                                   final CaseCheck theCase) {
+        final List<VariableData> returnList = new ArrayList<>();
+
+        final char[] sortedVariables = variableNames.toCharArray();
+        Arrays.sort(sortedVariables);
+
+        for (final char ch : sortedVariables) {
+            if (theCase.check(ch)) {
+                if (getVariable(ch, returnList, false) != null) {
+                    continue;
+                }
+
+                VariableData thisEquation = getVariable(ch, variables, true);
```

Comment:
This method modifies the content of parameter "variables", which should not be done. At least there should be a comment on the method telling which params will be modified.

Reply:
Done